runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -173,6 +177,24 @@ public void checkSpots()
 		}
 	}
 
+	@Schedule(
+			period = 1,
+			unit = ChronoUnit.SECONDS
+	)
+	public void checkAmmo()
```

Comment:
Any reason why this doesn't just check the count whenever the cannonball is fired?